base/acme/src/org/dogtagpki/acme/ACMEHeader.java

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme;
+
+import java.net.URI;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Endi S. Dewata
+ */
+@JsonInclude(Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class ACMEHeader {
+
+    private String alg;
+    private JWK jwk;
```

Comment:
UGH, JWK/JWS is terrible to get right. Do you have that implemented yet? Are you using a helper library for it?

Reply:
In a later patch I'm generating JWK thumbprint to be used as account ID which is I think also used for ACME challenge. So far it's working fine with certbot. See the following code:
* https://github.com/edewata/pki/commit/cbc823aecc3462132cae3f1f2826ace34372c372#diff-a3185bddea60201699a80824778a9af4R50
* https://github.com/edewata/pki/commit/cbc823aecc3462132cae3f1f2826ace34372c372#diff-d9d69448fbf354ff2619a045507ec743R245

Reply:
Hmm ok. Thanks!